bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -552,6 +552,10 @@ public static boolean checkIsType(Type sourceType, Type targetType) {
 
     @Deprecated
     public static boolean checkIsType(Type sourceType, Type targetType, List<TypePair> unresolvedTypes) {
+        if (checkIsNeverTypeOrStructureTypeWithARequiredNeverMember(sourceType)) {
```

Comment:
I think we should move this to after the same type check, since the chances of this being true is less.